Source/RCellFinder_Zombiefied.cs

```diff
@@ -620,10 +601,9 @@ public static bool TryFindRandomSpotJustOutsideColony(IntVec3 root, Map map, Paw
                     return false;
                 }
                 int num = 0;
-                CellRect.CellRectIterator iterator = CellRect.CenteredOn(c, walkRadius).GetIterator();
-                while (!iterator.Done())
+                foreach (IntVec3 cell in CellRect.CenteredOn(c, walkRadius))
                 {
-                    Room room2 = iterator.Current.GetRoom(map);
+                    Room room2 = cell.GetRoom(map);
```

Comment:
I agree that what you did here looks fine. but this is decompiled Rimworld code so I wouldnt change it. There may be some weird reason for this being an iterator

Reply:
Twas actually the decompilation which pointed me towards it actually, I wouldn't've even messed with it otherwise.
```cs
[Obsolete("Use foreach on the cellrect instead")]
public CellRect.CellRectIterator GetIterator()
{
return new CellRect.CellRectIterator(this);
}
```

Reply:
Oke fair enough